src/core/types/message.rs

```diff
@@ -11,13 +11,26 @@ pub struct Message {
     pub date: i32,
     pub chat: Chat,
     #[serde(flatten)]
-    pub message_kind: MessageKind,
+    pub kind: MessageKind,
+}
+
+impl Message {
+    fn text(&self) -> Option<&str> {
+        if let MessageKind::CommonMessage {
+                media_kind: MediaKind::Text {
+                    ref text, ..
+                }, .. } = self.kind {
+            Some(text)
+        } else {
+            None
+        }
+    }
 }
 
 #[derive(Debug, Deserialize, PartialEq, Clone)]
 #[serde(untagged)]
 pub enum MessageKind {
-    IncomingMessage {
+    CommonMessage {
```

Comment:
I think that should be named just `Common` as other variants (`NewChatMembers` and not `NewChatMembersMessage`)
```suggestion
    Common {
```

Reply:
I'll fix it directly in `dev`.